peps/pep-8106.rst

```diff
@@ -0,0 +1,171 @@
+PEP: 8106
+Title: AnnotatedMeta for type checking of PEP 593 Annotated
+Author: Adrian Garcia Badaracco <adrian@adriangb.com>
+Sponsor: Jelle Zijlstra <jelle.zijlstra@gmail.com>
+Discussions-To: https://discuss.python.org/t/XYZ
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Created: 20-May-2024
+Python-Version: 3.13
+Post-History: 20-May-2024
+
+Abstract
+--------
+
+This PEP proposes a mechanism for integrating metadata in the :pep:`593` ``Annotated`` type
+into the type system.
+
+Motivation
+----------
+
+:pep:`593` introduced ``Annotated`` as a way to attach runtime metadata to types,
+however it does not provide any way to tie the metadata to the type itself.
+This can introduce type unsoundness in real world usage, even in examples similar
+to those in the PEP.
+For example, given a metadata like ``Int64()`` that might instruct how to serialize a field
+over the wire, one can write ``Annotated[float, Int64()]`` which is clearly unsound
+and could be type checked, but there is currently no mechanism to do so.
+
+Rationale
+---------
+
+This PEP adds ``AnnotatedMeta`` to the typing module, which is a generic class
+that when unsubstituted is equivalent to the type in ``Annotated``.
+
+For example::
+    from typing import AnnotatedMeta
+
+    class Int64(AnnotatedMeta[int]):
+        ...
+  
+    Annotated[int, Int64()]  # okay
+    Annotated[float, Int64()]  # error because AnnotatedMeta[float] != AnnotatedMeta[int]
+
+Motivating examples
```

Comment:
This should be part of the "Motivation" section.

Reply:
I moved both of these sub-sections to `Motivation` in eeed4061